Documentation/ceph-encryption.md

```diff
@@ -0,0 +1,225 @@
+  mon:
+    count: 3
+    allowMultiplePerNode: false
+  storage:
+    useAllNodes: false
+    useAllDevices: false
+    onlyApplyOSDPlacement: false
+    encryptedDevice: "true"  # ==> enable encryption
+    nodes:
+      < hosts and disks goes here >
+```
+
+2.  PVC based Ceph cluster: Rook's cloud native & recommended [Ceph cluster
+    deployment](https://github.com/rook/rook/blob/master/Documentation/ceph-cluster-crd.md#pvc-based-cluster)
+    where Ceph data is stored on volumes requested through StorageClasses & K8s.
+    Note: For PVC based clusters, [Rook will always use raw
+    mode](https://github.com/rook/rook/blob/2f850b6ae65ac632d594c091f6d3a8422bf3b2ef/pkg/daemon/ceph/osd/volume.go#L483).
+    In these cases when encryption is specified, rook takes it upon itself to
+    generate the encryption key and set the required environment variable
+    expected by ceph-volume. It also saves the generated key as either a K8s
+    secret or through a KMS.
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: local-storage
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
+---
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: local0-1
+spec:
+  storageClassName: local-storage
+  capacity:
+    storage: 10Gi
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  volumeMode: Block
+  local:
+    path: /dev/sdc
+  nodeAffinity:
+    required:
+        ...
+---
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph
+spec:
+  dataDirHostPath: /var/lib/rook
+  mon:
+    count: 3
+    allowMultiplePerNode: false
+    volumeClaimTemplate:
+      spec:
+        storageClassName: local-storage
+        resources:
+          requests:
+            storage: 10Gi
+  cephVersion:
+    image: quay.io/ceph/ceph:v16.2.6
+    allowUnsupported: false
+  mgr:
+    count: 1
+  storage:
+    storageClassDeviceSets:
+      - name: set1
+        count: 1
+        portable: false
+        tuneDeviceClass: true
+        tuneFastDeviceClass: false
+        encrypted: true    # ==> enable encryption
+```
+
+## Customer requirements & encryption at rest
+There are scenarios when the managers of the Ceph cluster are different than the
+consumers of the cluster. In those scenarios, encryption at rest can be a legal
+and compliance requirement, that includes key rotation support and the ability
+for customers to bring their own encryption keys.
+
+This introduces new feature requirements into rook/ceph:
+1. Support for key rotation.
+2. Support for BYOK for encryption.
+
+## Key Rotation
+As of today, neither Ceph nor Rook support a flag or easy mechanism to rotate a
+Ceph key. Key rotation is feasible through generating a new key and adding it to
+the LUKS header using cryptsetup. There are 2 different approaches to this:
+-  Updating the LUKS key: LUKS header can be updated to change the key, ex:
```

Comment:
My initial thoughts was to use `luksChangeKey` just in case the keyslots are all used then a `luksAddKey` will fail. I can look more into how both are implemented. 
+1 on KEK rotation.